src/peft/tuners/lora/layer.py

```diff
@@ -125,9 +135,21 @@ def update_layer(
         if use_dora:
             self.dora_init(adapter_name)
             self.use_dora[adapter_name] = True
+
         else:
             self.use_dora[adapter_name] = False
 
+        if use_wlora:
+            # initialize wlora_weights
+            if not self.wlora_weights:
+                self.wlora_weights = nn.ParameterDict()
+            self.wlora_weights[adapter_name] = nn.Parameter(torch.tensor([1.0]))
+            # add `wlora_weights`` to the list of learnable parameters
+            self.adapter_layer_names = self.adapter_layer_names[:] + ("wlora_weights",)
+            self.use_wlora[adapter_name] = True
```

Comment:
I wonder if we could not automatically deactivate the gradients for the normal LoRA weights if we use wlora. So basically: `self.lora_A[adapter_name].requires_grad_(False)` etc. WDYT?